geospace-lib/src/main/scala/com/socrata/geospace/lib/regioncache/RegionCache.scala

```diff
@@ -48,8 +48,11 @@ abstract class RegionCache[T](maxEntries: Int = 100) //scalastyle:ignore
 
   // There's a bug in the scala-metrics library - metrics.gauge doesn't check if
   // the metric already exists before trying to registering it again.
-  // Because of this, unit tests fail unless we give the gauge metric a unique scope per instance
-  val cacheSizeGauge = metrics.gauge("num-entries", System.currentTimeMillis.toString) { cache.size }
+  // Because of this, unit tests fail unless we check for existence and skip the gauge.
+  if (!metrics.registry.getNames.contains("num-entries")) {
+    metrics.gauge("num-entries") { cache.size }
+  }
```

Comment:
Gauges only report the value of the closure on demand, which in this case is the graphite and log reporting frequency. 

-Evan
"Never doubt that a small group of thoughtful, committed citizens can change the world" - M. Mead

> On Apr 28, 2015, at 8:45 PM, Chi Tang <notifications@github.com> wrote:
> 
> In geospace-lib/src/main/scala/com/socrata/geospace/lib/regioncache/RegionCache.scala:
> 
> > @@ -48,8 +48,11 @@ abstract class RegionCache[T](maxEntries: Int = 100) //scalastyle:ignore
> >  
> >    // There's a bug in the scala-metrics library - metrics.gauge doesn't check if
> >    // the metric already exists before trying to registering it again.
> > -  // Because of this, unit tests fail unless we give the gauge metric a unique scope per instance
> > -  val cacheSizeGauge = metrics.gauge("num-entries", System.currentTimeMillis.toString) { cache.size }
> > +  // Because of this, unit tests fail unless we check for existence and skip the gauge.
> > +  if (!metrics.registry.getNames.contains("num-entries")) {
> > +    metrics.gauge("num-entries") { cache.size }
> > +  }
> I believe that gauge is updated at a constant interval specified in configuration. I don't think gauges detect change and capture value.
> 
> Also addressed thread safety problem.
> 
> —
> Reply to this email directly or view it on GitHub.
>